include/swift/AST/DiagnosticsSema.def

```diff
@@ -3016,6 +3016,9 @@ ERROR(enum_raw_type_not_equatable,none,
 ERROR(enum_raw_type_nonconforming_and_nonsynthable,none,
       "%0 declares raw type %1, but does not conform to RawRepresentable "
       "and conformance could not be synthesized", (Type, Type))
+ERROR(existential_enum_conformance_unsupported,none,
+      "%0 declares raw type %1, existential types cannot conform to protocol RawRepresentable",
```

Comment:
I think `but` is missing.

Reply:
How about `Enum raw type %1 cannot conform to protocol 'RawRepresentable'`?

Reply:
I wonder if there is a way to phrase it in a way that gives an indication why it cannot conform...

Reply:
Hmm, maybe we should add an attached note that we can also use for `type_cannot_conform`. I do think it'd be great to provide a little bit of information about why `any` types can't conform to protocols

Reply:
Yeah, note should work fine!

Reply:
I think the current diagnosis is adequate in this respect:
```
error: raw type 'any P' is not expressible by a string, integer, or floating-point literal
enum E: any P {
        ^
error: 'E' declares raw type 'any P', but does not conform to RawRepresentable and conformance could not be synthesized
enum E: any P {
        ^
error: RawRepresentable conformance cannot be synthesized because raw type 'any P' is not Equatable
enum E: any P {

```

This fails because a raw type was specified but the conformance of the enum to `RawRepresentable` cannot be synthesized, not because the existential raw type does not conform to `RawRepresentable`. The raw type can be anything in a custom conformance.

Perhaps we should ask if the user intended to conform to `P` instead?

Reply:
`: P` in this context is `: any P` it's not a conformance.

Reply:
To clarify are you suggesting that we simplify treat `: any P` as if `any` was written unintentionally then?

Reply:
Yes, I think it’s sensible to assume the user may have intended to conform to `P` if `any P` appears in an inheritance clause. Have I misinterpreted the PR description?

I am trying to make sense of this new diagnostic. How does the fact that an existential cannot conform to `RawRepresentable` explain why an existential raw type is rejected?

> `: P` in this context is `: any P` it's not a conformance.

Sorry, you’ve lost me there 😅

Reply:
When it comes to enums `:` syntax could be used for both raw value declaration and protocol conformance declaration, raw value declaration is supposed to conform to `RawRepresentable` protocol and match the type used on the right-hand side of the `=` of case statements. If I understand it correctly existential type couldn't be used as `raw type` because `RawRepresentable` has associated type requirements but if that's the case I guess it could be allowed to use primary associated types feature to work around that?

I wonder if the diagnostic here should be `cannot use existential type as raw type of enum %0` and provide a note that 1. `existential value of %0 cannot conform to 'RawRepresentible' protocol because it has associated type requirements`?

Reply:
@angela-laar ^

Reply:
> raw value declaration is supposed to conform to RawRepresentable protocol 

Wait, I think you have the wrong idea. We do not require the raw type to to conform to `RawRepresentable`. For example, `String` is a popular raw type, but does not conform. The only thing we ask of the raw type (if it is specified in the inheritance clause, as opposed to a custom conformance) is that it conforms to `ExpressibleBy(Integer|Float|String)Literal`.